examples/llm/main.py

```diff
@@ -29,7 +53,16 @@ def build_composer_model(cfg):
         raise ValueError(f'Not sure how to build model with name={cfg.name}')
 
 
+def build_dataloader(cfg, device_batch_size):
+    if cfg.name == 'text_denoising':
+        return build_text_denoising_dataloader(cfg, device_batch_size)
+    return _build_dataloader(cfg, device_batch_size)
```

Comment:
oh this is kinda obscure naming... should we augment the `_build_dataloader` inside `builders.py` so that it can switch on `cfg.name`?

Reply:
I'm fine with this, but it will add a very non-common (i.e. specific to LLM) dataloader in the common dataloader builder.

Reply:
Oh i see what you mean... hmm well if you think that the denoising dataloader won't be used elsewhere only here, perhaps we can just do:

```
from examples.common.text_data import build_text_dataloader
...
 if cfg.name == 'text_denoising':
        return build_text_denoising_dataloader(cfg, device_batch_size)
elif cfg.name == 'text':
    return return build_text_dataloader(cfg, device_batch_size)
else:
  raise ValueError(...)
```

just to make it easier to grok what's going on and click through to the relevant dataloader code